js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+
+**Multie Owner Authorization**
+
+Work in progress.
+
+**Static Group Authorization**
+
+```
+type Post @model @auth(rules: [{allow: groups, groups: ["Admin"]}]) {
+  id: ID!
+  title: String!
+  groups: String 
+}
+```
+
+Static group auth is simpler than the others. The generated resolvers would be protected like so:
+
+- `Mutation.createX`: Verify the requesting user has a valid credential and that `ctx.identity.claims.get("cognito:groups")` contains the **Admin** group. If it does not, fail`.
+- `Mutation.updateX`: Verify the requesting user has a valid credential and that `ctx.identity.claims.get("cognito:groups")` contains the **Admin** group. If it does not, fail.
+- `Mutation.deleteX`: Verify the requesting user has a valid credential and that `ctx.identity.claims.get("cognito:groups")` contains the **Admin** group. If it does not, fail.
+- `Query.getX`: Verify the requesting user has a valid credential and that `ctx.identity.claims.get("cognito:groups")` contains the **Admin** group. If it does not, fail.
+- `Query.listX`: Verify the requesting user has a valid credential and that `ctx.identity.claims.get("cognito:groups")` contains the **Admin** group. If it does not, fail.
+
+**Dynamic Group Authorization**
+
+```
+type Post @model @auth(rules: [{allow: groups, groupsField: "groups"}]) {
+  id: ID!
+  title: String!
+  groups: String 
+}
+```
+
+the generated resolvers would be protected like so:
+
+- `Mutation.createX`: Verify the requesting user has a valid credential and that it contains a claim to atleast one group passed to the query in the `$ctx.args.input.groups` argument.
+- `Mutation.updateX`: Update the condition expression so that the DynamoDB `UpdateItem` operation only succeeds if the record's **groups** attribute contains at least one of the caller's claimed groups via `ctx.identity.claims.get("cognito:groups")`.
+- `Mutation.deleteX`: Update the condition expression so that the DynamoDB `DeleteItem` operation only succeeds if the record's **groups** attribute contains at least one of the caller's claimed groups via `ctx.identity.claims.get("cognito:groups")`
+- `Query.getX`: In the response mapping template verify that the result's **groups** attribute contains at least one of the caller's claimed groups via `ctx.identity.claims.get("cognito:groups")`.
+- `Query.listX`: In the response mapping template filter the result's **items** such that only items with a **groups** attribute that contains at least one of the caller's claimed groups via `ctx.identity.claims.get("cognito:groups")`.
+
+
+### @connection
+
+The `@connection` directive enables you to specify relationships between `@model` object types.
+Currently, this supports one-to-one, one-to-many, and many-to-one relationships. An error
+is thrown if you try to configure a many-to-many relationship.
```

Comment:
@mikeparisstuff I thought there was a way that customers could "fake" many to many with 2x one to many though? If so we should document that.